prestashopv1/plugin/modules/indodana/tools/IndodanaTools.php

```diff
@@ -0,0 +1,166 @@
+                * $product['quantity'];
+            $deficit += self::getDeficit($product);
+        }
+
+        return round($tax) + $deficit;
+    }
+
+    public static function getProducts($cart)
+    {
+        $items = [];
+        $products = $cart->getProducts();
+        foreach ($products as $product) {
+            $items[] = [
+                'id' => $product['reference'],
+                'name' => $product['name'],
+                'price' => round(self::getPriceWithoutReductionWithoutTax($product)),
+                'quantity' => $product['quantity'],
+                'category'  => IndodanaCommon\IndodanaConstant::DEFAULT_ITEM_CATEGORY,
+            ];
+        }
+
+        return $items;
+    }
+
+    public static function getCustomerDetails($orderId)
+    {
+        $orderDetails = new Order((int) $orderId);
+        $customer = new Customer((int) $orderDetails->id_customer);
+        $invoiceDetails = new Address((int) $orderDetails->id_address_invoice);
+
+        return [
+            'firstName' => $customer->firstname,
+            'lastName' => $customer->lastname,
+            'email' => $customer->email,
+            'phone' => $invoiceDetails->phone
+        ];
+    }
+
+    public static function getBillingAddress($orderId)
+    {
+        $orderDetails = new Order((int) $orderId);
+        $invoiceDetails = new Address((int) $orderDetails->id_address_invoice);
+
+        return [
+            'firstName' => $invoiceDetails->firstname,
+            'lastName' => $invoiceDetails->lastname,
+            'address' => $invoiceDetails->address1 . ' ' . $invoiceDetails->address2,
+            'city' => $invoiceDetails->city,
+            'postalCode' => $invoiceDetails->postcode,
+            'phone' => $invoiceDetails->phone,
+            'countryCode' => Configuration::get('INDODANA_STORE_COUNTRY_CODE')
+        ];
+    }
+
+    public static function getDeliveryAddress($orderId)
+    {
+        $orderDetails = new Order((int) $orderId);
+        $deliveryDetails = new Address((int)($orderDetails->id_address_delivery));
+
+        return [
+            'firstName' => $deliveryDetails->firstname,
+            'lastName' => $deliveryDetails->lastname,
+            'address' => $deliveryDetails->address1 . ' ' . $deliveryDetails->address2,
+            'city' => $deliveryDetails->city,
+            'postalCode' => $deliveryDetails->postcode,
+            'phone' => $deliveryDetails->phone,
+            'countryCode' => Configuration::get('INDODANA_STORE_COUNTRY_CODE')
+        ];
+    }
+
+    public static function getTotalProduct($product)
+    {
+        return round(self::getPriceWithoutReductionWithoutTax($product) * $product['quantity']);
+    }
+
+    /**
+     * sometimes when rounding up/down numbers there will be deficit
+     * so that the total price calculation does not match
+     */
+    public static function getDeficit($product)
```

Comment:
`getDeficit` only used by `getTax`. Please set this to `private` (I believe you can `private static function`, CMIIW though).

If can, please do this on some functions within this file that supposed to be `private`

Reply:
Okay, I'll change it to private